### ISSUE TITLE
[1.4.230] Microphone is quickly overloaded by whistling/crackling

### ISSUE THREAD

**Issue by mikebdp2** [mumble-voip/mumble#5506]
### Description

_This is a follow-up for [Mumble v1.3.4 microphone problem](https://github.com/mumble-voip/mumble/issues/5475)._

Even the slightest noise seems to initiate a chain reaction. What seems to be happening:

- microphone gets the sound, speakers play it, microphone picks the sound which gets played by speakers even louder, and so on. Quickly, the speakers are playing the loud whistling/crackling which is also the only thing a microphone is now getting.

This happens during the Audio Wizard setup and prevents from finding the optimal positions of the sliders. A temporary solution is to temporarily disable the speakers while you are setting up the microphone.

### Steps to reproduce

I launch Mumble and open the Audio Wizard. on **Device Selection** screen I see these devices: _(click a spoiler)_ <details><summary><i>ALSA :</i></summary>[sysdefault:CARD=PinePhone] PinePhone, 1c22c00.dai-sun8i-codec-aif1 sun8i-codec-aif1-0 Default Audio Device<br>[sysdefault:CARD=sun9ihdmi] sun9i-hdmi, SUN9I-HDMI PCM i2s-hifi-0 Default Audio Device<br>[default] Default ALSA Output (currently PulseAudio Sound Server)<br>[null] Discard all samples (playback) or generate zero samples (capture)<br>[pulse] PulseAudio Sound Server<br></details><details><summary><i>JACK :</i></summary>Mono<br>Stereo<br></details><details><summary><i>PulseAudio :</i></summary>Default Input<br>Default Output<br>PinePhone Headset Microphone + Internal Microphone<br>PinePhone Headset + Internal Earpiece + Internal speaker<br>sun9i-hdmi Stereo<br></details>

It seems logical to choose:
- "Pulseaudio ==> PinePhone Headset Microphone + Internal Microphone" as the input device
- "Pulseaudio ==> PinePhone Headset + Internal Earpiece + Internal speaker" as the output device.

There are also "Use echo cancellation", "Enable positional audio" and "Attenuate applications while other users talk" - which don't seem to affect the subsequent "sound overload" problem.

"Next" button brings me to **Device tuning** for speakers, which seems to work fine. One more "Next" - to **Volume tuning** screen, where the "sound overload" problem appears. No matter the position of slider, the sound quickly gets to high overloaded levels. On the next **Voice activity detection** screen, there are the following options:
- Push to talk
- Raw amplitude from input
- Signal-to-Noise ratio

Somehow, switching from "Raw amplitude from input" to "Push to talk" _(without configuring a keyboard shortcut, which I couldn't do)_ - temporarily fixes the mic input, as evidenced by not-overloaded volume indicator below which shows a normal response to the incoming sound and doesn't get overloaded with the time.

### Mumble version

1.4.230

### Mumble component

Client

### OS

Linux

### Reproducible?

Yes

### Additional information

I'm using [PostmarketOS](https://postmarketos.org/) which is based on Alpine Linux: it provides **[Mumble v1.4.230](https://gitlab.alpinelinux.org/alpine/aports/-/tree/master/community/mumble)** ARM aarch64 package built with these options:

- **[APKBUILD](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/mumble/APKBUILD)**
- [murmur.initd](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/mumble/murmur.initd)
- [murmur.pre-install](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/mumble/murmur.pre-install)
- [libgen.patch](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/mumble/libgen.patch)
- [no-rnnoise.patch](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/mumble/no-rnnoise.patch)
- [poll.patch](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/mumble/poll.patch)

**Comment by Krzmbrzl** [comment]
Are you using the same audio Backend for input and output?
Foes the same issue occur if you are having a normal conversation or is this an issue exclusive to the audio wizard?

**Comment by mikebdp2** [comment]
@Krzmbrzl Yes, I'm using a Pulseaudio as audio backend both for input and output, as indicated above at _"It seems logical to choose"_ part
>  Does the same issue occur if you are having a normal conversation or is this an issue exclusive to the audio wizard?

This issue is _almost_ exclusive to the Audio Wizard: it'll also happen if a second device _(i.e. Android with Mumla)_ - with which I'm having a conversation - is in the same room, so close enough to Pinephone and its' mic. I don't know if this should be expected

**Comment by Krzmbrzl** [comment]
Using a separate device in the same room is definitely expected. There is absolutely no way to distinguish this from proper audio input.

The way echo cancellation works is by analyzing the audio output and comparing that with the current audio input (and filtering out that part that have been encountered in the output already).

The issue only appearing in the audio wizard makes me think that in this context we somehow don't apply echo cancellation :thinking:

**Comment by mikebdp2** [comment]
@Krzmbrzl
> The issue only appearing in the audio wizard makes me think that in this context we somehow don't apply echo cancellation

In the "same room" example above, where I also encountered this problem, it was happening with the echo cancellation enabled - which makes me think, maybe it's not working good enough? I just tested with a regular phone call and this problem is also happening but at a smaller degree

**Comment by Krzmbrzl** [comment]
Please re-read my previous post. Especially this part:
> There is absolutely no way to distinguish this from proper audio input.

In short: There is no way on earth that echo cancellation can fix this situation for you, because it is lacking the necessary information
